dicetable.py

```diff
@@ -0,0 +1,327 @@
+            if multiplier == 0:
+                count+=1
+            else:
+                if x<10:
+                    print ' '+str(x)+': '+multiplier*'x'
+                else:
+                    print str(x)+': '+multiplier*'x'
+        if count !=0:
+            outrange = count/2-1
+            start = self.min()
+            finish = self.max()
+            outbottom = str(start)+' - '+str(start+outrange)
+            outtop = str(finish-outrange)+' - '+str(finish)
+        print 'each x represents '+(divstring)+' occurences' 
+        if count != 0:
+            print 'not included: '+outbottom+' and '+outtop
+        print self
+    
+    
+            
+            
+def scinote(num):
+    '''returns str of a rounded INT in sci notation. 
+    if you int(the string) it's legal input for python'''
+    string = str(num)
+    power = str(len(string)-1)
+    digits = string[0]+'.'+string[1:4]
+    return digits+'e+'+power
+
+
+def makealist():
+    '''take user input of a range of numbers
+    and outputs (list of numbers,input string)
+    a helper function for getstat'''
+    x = raw_input('enter number(s) you want stats of separated by commas or you can use a dash \nfor a range of numbers: ')
+    if x == 'q' or x == '':
+        return None,None
+    out = []
+    try:
+        for el in x.split(','):
+            if '-' in el:
+                start = int(el.split('-')[0])
+                end = int(el.split('-')[1])
+                if start <= end:
+                    for num in range(start,end+1):
+                        out.append(num)
+                else:
+                    for num in range(end,start+1):
+                        out.append(num)   
+            else:
+                out.append(int(el))
+        
+    except ValueError:
+        print
+        print 'ooops.  looks like someone can\'t follow directions.'
+        print 'your prize is a list that includes 0 and some of your elements.'
+        print 'here ya go.'
+        print
+        out.append(0)
+    return out,x
+
+
+#i need to modify the overflowerror, maybe use a function like scinote. 
+#right now, since overflowerror just uses int the chance will say 1 in 2 when pct is 34-50
+def getstat(dtable):
+        '''returns the stats on lst of elements in self'''
+        totalval = 0
+        tabletotal = dtable.dsize**dtable.totaldice
+        #below takes the values from makealist to get lst for function and 
+        # str to print the range of values
+        lst,lststring = makealist()
+        if lst == None:
+            return None
+        
+        for el in lst:
+            totalval += dtable.table.get(el,0)
+        if totalval == 0:
+            print 'you get nothing!  good day, sir!'
+            return 0
+        
+        
+        else:
+            
+            try:
+                chance = round(float(tabletotal)/totalval,3)
+            except OverflowError:
+                #!!personal reminder!!
```

Comment:
Having notes to future you is good practice. These should be TODOs https://google.github.io/styleguide/pyguide.html#TODO_Comments